pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,450 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import {
+    block_name, get_active_usage, teardown_active_usage,
+    is_mounted_synch, get_partitions
+} from "./utils.js";
+import {
+    existing_passphrase_fields, init_existing_passphrase,
+    request_passphrase_on_error_handler
+} from "./crypto-keyslots.jsx";
+import {
+    dialog_open, SizeSlider, BlockingMessage, TeardownMessage,
+    init_active_usage_processes
+} from "./dialog.jsx";
+import { std_reply } from "./stratis-utils.js";
+
+const _ = cockpit.gettext;
+
+function lvol_or_part_and_fsys_resize(client, lvol_or_part, size, offline, passphrase) {
+    let fsys;
+    let crypto_overhead;
+    let vdo;
+    let stratis_bdev;
+    let orig_size;
+    let block;
+
+    if (lvol_or_part.iface == "org.freedesktop.UDisks2.LogicalVolume") {
+        orig_size = lvol_or_part.Size;
+        block = client.lvols_block[lvol_or_part.path];
+        if (!block)
+            return lvol_or_part.Resize(size, { });
+    } else {
+        orig_size = lvol_or_part.Size;
+        block = client.blocks[lvol_or_part.path];
+    }
+
+    console.log("+", orig_size, block);
+
+    const crypto = client.blocks_crypto[block.path];
+    if (crypto) {
+        const cleartext = client.blocks_cleartext[block.path];
+        if (!cleartext)
+            return;
+        fsys = client.blocks_fsys[cleartext.path];
+        vdo = client.legacy_vdo_overlay.find_by_backing_block(cleartext);
+        stratis_bdev = client.blocks_stratis_blockdev[cleartext.path];
+        if (crypto.MetadataSize !== undefined)
+            crypto_overhead = crypto.MetadataSize;
+        else
+            crypto_overhead = block.Size - cleartext.Size;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19385-20230925-111150-44391ef2-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)